gammapy/datasets/map.py

```diff
@@ -2714,28 +2737,50 @@ def update(self, exposure, psf, edisp, geom, mask):
                     geom = geom.to_wcs_geom()
 
                 self.psf = psf.get_psf_kernel(
-                    position=self.model.position,
-                    geom=geom,
-                    containment=PSF_CONTAINMENT
+                    position=self.model.position, geom=geom, containment=PSF_CONTAINMENT
                 )
-
+     
         if self.evaluation_mode == "local":
             self.contributes = self.model.contributes(mask=mask, margin=self.psf_width)
 
             if self.contributes:
                 self.exposure = exposure.cutout(
-                    position=self.model.position,
-                    width=self.cutout_width,
-                    odd_npix=True
+                    position=self.model.position, width=self.cutout_width, odd_npix=True
                 )
         else:
             self.exposure = exposure
 
+        if self.contributes:
+            self.update_spatial_oversampling_factor(self.geom)
+
         self._compute_npred.cache_clear()
         self._compute_flux_spatial.cache_clear()
         self._computation_cache = None
         self._cached_parameter_previous = None
 
+    def update_spatial_oversampling_factor(self, geom):
+        """Update spatial oversampling_factor for model evaluation"""
+        if (
```

Comment:
OK, why not. Bur there is some ambiguity on the nature of the bin (spatial or non-spatial)

Reply:
I see, in general one could have the same logic with optimised bin sizes for spectral models. In practice it's probably much less relevant. In case we extend the system in future one could still return e.g. a dict...